### ISSUE TITLE
dials.integrate: output appears to be somewhat nonsensical 

### ISSUE THREAD

**Issue by graeme-winter** [dials/dials#1318]
Examples: counts images from 0...n-1 not 1...n (output should be in people numbers)

Summary table the same

Block summary off-by-1:

```
+-----+---------+--------------+------------+--------------+------------+-----------------+
|   # |   Group |   Frame From |   Frame To |   Angle From |   Angle To |   # Reflections |
|-----+---------+--------------+------------+--------------+------------+-----------------|
|   0 |       0 |            0 |         33 |        -0.15 |        4.8 |             107 |
+-----+---------+--------------+------------+--------------+------------+-----------------+
```

i.e. the start angle was 0, so the block is 0...4.95°

Looking at this in: https://github.com/dials/dials/tree/integration-output-people-numbers

**Comment by graeme-winter** [comment]
<- like "owait this could be something weird I have done"

**Comment by graeme-winter** [comment]
OK, some of this is my fault, other bits are real (processing set with main sequence dials) 

```
Processing reflections in the following blocks of images:

 block_size: 20 frames

+-----+---------+--------------+------------+--------------+------------+-----------------+
|   # |   Group |   Frame From |   Frame To |   Angle From |   Angle To |   # Reflections |
|-----+---------+--------------+------------+--------------+------------+-----------------|
|   0 |       0 |            0 |         20 |          0   |        3   |              47 |
|   1 |       0 |           10 |         30 |          1.5 |        4.5 |              35 |
|   2 |       0 |           20 |         40 |          3   |        6   |              36 |
|   3 |       0 |           30 |         50 |          4.5 |        7.5 |              30 |
|   4 |       0 |           40 |         60 |          6   |        9   |              42 |
|   5 |       0 |           50 |         70 |          7.5 |       10.5 |              29 |
|   6 |       0 |           60 |         80 |          9   |       12   |              31 |
|   7 |       0 |           70 |         90 |         10.5 |       13.5 |              48 |
|   8 |       0 |           80 |        100 |         12   |       15   |              38 |
|   9 |       0 |           90 |        110 |         13.5 |       16.5 |              34 |
|  10 |       0 |          100 |        120 |         15   |       18   |              41 |
|  11 |       0 |          110 |        130 |         16.5 |       19.5 |              33 |

```

=> frame from 1 to N not 0...N

```
 Frames: 0 -> 20

 Number of reflections
  Partial:     5
  Full:        42
  In ice ring: 0
  Total:       47

 The following histogram shows the number of reflections predicted
 to have all or part of their intensity on each frame.

 1  [6 ]: *******************
 2  [10]: *******************************
 3  [10]: *******************************
 4  [9 ]: ****************************
 5  [11]: ***********************************
 6  [12]: **************************************
 7  [14]: ********************************************
 8  [18]: *********************************************************
 9  [20]: ***************************************************************
 10 [22]: **********************************************************************
 11 [21]: ******************************************************************
 12 [22]: **********************************************************************
 13 [20]: ***************************************************************
 14 [17]: ******************************************************
 15 [14]: ********************************************
 16 [9 ]: ****************************
 17 [7 ]: **********************
 18 [3 ]: *********
 19 [1 ]: ***

 Modelled     0 /     3 reflection profiles on image 2
 Modelled     1 /     1 reflection profiles on image 3
 Modelled     2 /     2 reflection profiles on image 4
 Modelled     2 /     3 reflection profiles on image 5
 Modelled     0 /     1 reflection profiles on image 6
 Modelled     2 /     3 reflection profiles on image 7
 Modelled     2 /     2 reflection profiles on image 8
 Modelled     3 /     3 reflection profiles on image 9
 Modelled     3 /     3 reflection profiles on image 10
 Modelled     5 /     5 reflection profiles on image 11
 Modelled     4 /     4 reflection profiles on image 12
 Modelled     3 /     3 reflection profiles on image 13
 Modelled     5 /     5 reflection profiles on image 14
 Modelled     2 /     2 reflection profiles on image 15
 Modelled     4 /     4 reflection profiles on image 16
 Modelled     2 /     2 reflection profiles on image 17
 Modelled     1 /     1 reflection profiles on image 18
 Beginning modelling job 1
```

not really helpful - no reflections on frame (0,1?) 

Then later 

```
Processing reflections in the following blocks of images:

 block_size: 22 frames

+-----+---------+--------------+------------+--------------+------------+-----------------+
|   # |   Group |   Frame From |   Frame To |   Angle From |   Angle To |   # Reflections |
|-----+---------+--------------+------------+--------------+------------+-----------------|
|   0 |       0 |            0 |         22 |         0    |       3.3  |             320 |
|   1 |       0 |           11 |         33 |         1.65 |       4.95 |             147 |
|   2 |       0 |           22 |         44 |         3.3  |       6.6  |             156 |
|   3 |       0 |           33 |         55 |         4.95 |       8.25 |             160 |
|   4 |       0 |           44 |         66 |         6.6  |       9.9  |             158 |
|   5 |       0 |           55 |         77 |         8.25 |      11.55 |             145 |
|   6 |       0 |           66 |         88 |         9.9  |      13.2  |             169 |
|   7 |       0 |           77 |         99 |        11.55 |      14.85 |             150 |
|   8 |       0 |           88 |        110 |        13.2  |      16.5  |             160 |
|   9 |       0 |           99 |        121 |        14.85 |      18.15 |             174 |
|  10 |       0 |          110 |        132 |        16.5  |      19.8  |             155 |
|  11 |       0 |          121 |        143 |        18.15 |      21.45 |             144 |
|  12 |       0 |          132 |        154 |        19.8  |      23.1  |             150 |

```

yup this wants fixing then 

```
 The following histogram shows the number of reflections predicted
 to have all or part of their intensity on each frame.

 1  [136]: *********************************************************************
 2  [74 ]: *************************************
 3  [78 ]: ***************************************
 4  [75 ]: **************************************
 5  [73 ]: *************************************
 6  [72 ]: ************************************
 7  [73 ]: *************************************
 8  [76 ]: **************************************
 9  [82 ]: *****************************************
 10 [80 ]: ****************************************
 11 [84 ]: ******************************************
 12 [83 ]: ******************************************
 13 [84 ]: ******************************************
 14 [78 ]: ***************************************
 15 [74 ]: *************************************
 16 [59 ]: *****************************
 17 [52 ]: **************************
 18 [38 ]: *******************
 19 [22 ]: ***********
 20 [6  ]: ***
 21 [6  ]: ***
 22 [5  ]: **

 Integrated    62 (sum) +     0 (prf) /   77 reflections on image 0
 Integrated     7 (sum) +     0 (prf) /   10 reflections on image 1
 Integrated     9 (sum) +     7 (prf) /   12 reflections on image 2
 Integrated    13 (sum) +     9 (prf) /   16 reflections on image 3
 Integrated     9 (sum) +     9 (prf) /   11 reflections on image 4
 Integrated    13 (sum) +    13 (prf) /   16 reflections on image 5
 Integrated     7 (sum) +     7 (prf) /   10 reflections on image 6
 Integrated     9 (sum) +     9 (prf) /   12 reflections on image 7
 Integrated    14 (sum) +    14 (prf) /   17 reflections on image 8
 Integrated     9 (sum) +     9 (prf) /    9 reflections on image 9
 Integrated    10 (sum) +    10 (prf) /   11 reflections on image 10
 Integrated     9 (sum) +     9 (prf) /   12 reflections on image 11
 Integrated    13 (sum) +    13 (prf) /   18 reflections on image 12
 Integrated    13 (sum) +    13 (prf) /   15 reflections on image 13
 Integrated    14 (sum) +    14 (prf) /   15 reflections on image 14
 Integrated     7 (sum) +     7 (prf) /    7 reflections on image 15
 Integrated    13 (sum) +    13 (prf) /   14 reflections on image 16
 Integrated     9 (sum) +     9 (prf) /   16 reflections on image 17
 Integrated    15 (sum) +    15 (prf) /   16 reflections on image 18
 Integrated     1 (sum) +     1 (prf) /    1 reflections on image 20
 Integrated     3 (sum) +     3 (prf) /    5 reflections on image 21
 Beginning integration job 1
```

1st bit counts from 1...N second from 0...(N-1) 🤦‍♂️ 

Then at the end 

```
 The following histogram shows the number of reflections predicted
 to have all or part of their intensity on each frame.

 2388 [5  ]: **
 2389 [13 ]: ******
 2390 [22 ]: **********
 2391 [45 ]: **********************
 2392 [56 ]: ***************************
 2393 [69 ]: **********************************
 2394 [81 ]: ****************************************
 2395 [80 ]: ***************************************
 2396 [83 ]: *****************************************
 2397 [81 ]: ****************************************
 2398 [79 ]: ***************************************
 2399 [81 ]: ****************************************
 2400 [135]: *******************************************************************

 Integrated     3 (sum) +     3 (prf) /    3 reflections on image 2392
 Integrated    15 (sum) +    15 (prf) /   16 reflections on image 2393
 Integrated    14 (sum) +    14 (prf) /   18 reflections on image 2394
 Integrated    12 (sum) +    12 (prf) /   15 reflections on image 2395
 Integrated    10 (sum) +    10 (prf) /   13 reflections on image 2396
 Integrated    12 (sum) +    12 (prf) /   14 reflections on image 2397
 Integrated    11 (sum) +    11 (prf) /   12 reflections on image 2398
 Integrated   107 (sum) +    35 (prf) /  135 reflections on image 2399

 Summary vs image number
+------+---------+----------+----------+----------+---------+---------+---------+-------+----------+----------+----------+-----------+
|   ID |   Image |   # full |   # part |   # over |   # ice |   # sum |   # prf |   Ibg |   I/sigI |   I/sigI |   CC prf |   RMSD XY |
|      |         |          |          |          |         |         |         |       |    (sum) |    (prf) |          |           |
|------+---------+----------+----------+----------+---------+---------+---------+-------+----------+----------+----------+-----------|
|    0 |       0 |        0 |      106 |        0 |       0 |      84 |      10 |  0.03 |     1.32 |     8.37 |     0.75 |      1.82 |
|    0 |       1 |        3 |       10 |        0 |       0 |      10 |      10 |  0.03 |     9.87 |    10.44 |     0.7  |      0.42 |
|    0 |       2 |        9 |        5 |        0 |       0 |      13 |      12 |  0.03 |     5.06 |     5.44 |     0.59 |      0.77 |
|    0 |       3 |       14 |        0 |        0 |       0 |      11 |      11 |  0.02 |     6.52 |     6.41 |     0.59 |      0.71 |
|    0 |       4 |       10 |        2 |        0 |       0 |       5 |       5 |  0.02 |     2.75 |     2.7  |     0.43 |      0.48 |
|    0 |       5 |       13 |        0 |        0 |       0 |      11 |      11 |  0.03 |     4.81 |     4.9  |     0.58 |      0.6  |
|    0 |       6 |       13 |        0 |        0 |       0 |      11 |      11 |  0.02 |     5.04 |     4.97 |     0.6  |      0.69 |
|    0 |       7 |        7 |        0 |        0 |       0 |       6 |       6 |  0.05 |    20.6  |    20.5  |     0.8  |      0.62 |
|    0 |       8 |       15 |        0 |        0 |       0 |      14 |      14 |  0.02 |     8.3  |     8.06 |     0.62 |      0.53 |
|    0 |       9 |       11 |        0 |        0 |       0 |       8 |       8 |  0.03 |     7.25 |     7.03 |     0.67 |      0.29 |
|    0 |      10 |       20 |        0 |        0 |       0 |      17 |      17 |  0.02 |     6.07 |     5.94 |     0.61 |      0.82 |
|    0 |      11 |       17 |        0 |        0 |       0 |      14 |      14 |  0.03 |     9.19 |     9.16 |     0.68 |      0.48 |
|    0 |      12 |       13 |        0 |        0 |       0 |      11 |      11 |  0.02 |     4.6  |     4.39 |     0.57 |      0.83 |
|    0 |      13 |        8 |        5 |        0 |       0 |      11 |      10 |  0.02 |     4.16 |     5.12 |     0.58 |      0.86 |
|    0 |      14 |       17 |        2 |        0 |       0 |      17 |      17 |  0.03 |    10.91 |    10.79 |     0.64 |      0.42 |
|    0 |      15 |       14 |        0 |        0 |       0 |       9 |       9 |  0.03 |     5.63 |     5.75 |     0.63 |      0.41 |
|    0 |      16 |       12 |        0 |        0 |       0 |      10 |      10 |  0.04 |    14.34 |    14.27 |     0.78 |      0.3  |
|    0 |      17 |       14 |        0 |        0 |       0 |       9 |       9 |  0.04 |    16.6  |    16.63 |     0.65 |      0.52 |
|    0 |      18 |        9 |        2 |        0 |       0 |      11 |      11 |  0.03 |    11.22 |    11.56 |     0.61 |      0.62 |

```

seems like counting from 0, 1 depends on 🎲  roll

**Comment by graeme-winter** [comment]
```
 1652 [3 ]: **
 1653 [3 ]: **
 1654 [7 ]: *****
 1655 [23]: *****************
 1656 [38]: ****************************
 1657 [55]: *****************************************
 1658 [70]: ****************************************************
 1659 [82]: *************************************************************
 1660 [90]: ********************************************************************
 1661 [85]: ****************************************************************
 1662 [87]: *****************************************************************
 1663 [81]: *************************************************************
 1664 [75]: ********************************************************
 1665 [62]: **********************************************
 1666 [53]: ****************************************
 1667 [40]: ******************************
 1668 [27]: ********************
 1669 [12]: *********
 1670 [7 ]: *****
 1671 [6 ]: ****
 1672 [6 ]: ****

 Integrated     2 (sum) +     2 (prf) /    2 reflections on image 1658
 Integrated     9 (sum) +     9 (prf) /   10 reflections on image 1659
 Integrated    16 (sum) +    16 (prf) /   18 reflections on image 1660
 Integrated    11 (sum) +     8 (prf) /   11 reflections on image 1661
 Integrated    13 (sum) +    11 (prf) /   17 reflections on image 1662
 Integrated    17 (sum) +    16 (prf) /   19 reflections on image 1663
 Integrated    18 (sum) +    18 (prf) /   21 reflections on image 1664
 Integrated     9 (sum) +     9 (prf) /    9 reflections on image 1665
 Integrated    12 (sum) +    12 (prf) /   13 reflections on image 1666
 Integrated    11 (sum) +    11 (prf) /   13 reflections on image 1667
 Integrated    10 (sum) +     9 (prf) /   15 reflections on image 1668
 Integrated     4 (sum) +     4 (prf) /    5 reflections on image 1669
 Integrated     1 (sum) +     1 (prf) /    1 reflections on image 1670
 Integrated     6 (sum) +     6 (prf) /    6 reflections on image 1672
 Beginning integration job 151

```

is also misleading because the histogram has images listed which are then (apparently) not integrated however they are, but not because the _background_ is what is contributing. Assert: this histogram should list the centroids or something? Also weird: the histogram is asymmetric w.r.t. the list below, since the former runs from earlier but ends at the same time? 🤔

**Comment by graeme-winter** [comment]
OK, we deffo have some residual asymmetry in here which does not make a lot of sense:

```

 Frames: 22 -> 44

 Number of reflections
  Partial:     8
  Full:        148
  In ice ring: 0
  Integrate:   156
  Total:       156

 The following histogram shows the number of reflections predicted
 to have all or part of their intensity on each frame.

 23 [1 ]: 
 24 [4 ]: ***
 25 [7 ]: ******
 26 [18]: ***************
 27 [30]: *************************
 28 [43]: *************************************
 29 [51]: ********************************************
 30 [61]: ****************************************************
 31 [74]: ***************************************************************
 32 [78]: *******************************************************************
 33 [77]: ******************************************************************
 34 [79]: ********************************************************************
 35 [81]: **********************************************************************
 36 [73]: ***************************************************************
 37 [71]: *************************************************************
 38 [57]: *************************************************
 39 [46]: ***************************************
 40 [32]: ***************************
 41 [19]: ****************
 42 [8 ]: ******
 43 [5 ]: ****
 44 [4 ]: ***

 Integrated     2 (sum) +     2 (prf) /    3 reflections on image 30
 Integrated     7 (sum) +     7 (prf) /   11 reflections on image 31
 Integrated    13 (sum) +    13 (prf) /   15 reflections on image 32
 Integrated    18 (sum) +    17 (prf) /   18 reflections on image 33
 Integrated    10 (sum) +    10 (prf) /   11 reflections on image 34
 Integrated    12 (sum) +    11 (prf) /   14 reflections on image 35
 Integrated    13 (sum) +    13 (prf) /   13 reflections on image 36
 Integrated    12 (sum) +    12 (prf) /   14 reflections on image 37
 Integrated     9 (sum) +     9 (prf) /   11 reflections on image 38
 Integrated    13 (sum) +    13 (prf) /   14 reflections on image 39
 Integrated    13 (sum) +    12 (prf) /   13 reflections on image 40
 Integrated     5 (sum) +     5 (prf) /   11 reflections on image 41
 Integrated     2 (sum) +     2 (prf) /    3 reflections on image 42
 Integrated     1 (sum) +     1 (prf) /    1 reflections on image 43
 Integrated     3 (sum) +     2 (prf) /    4 reflections on image 44
```

Ignoring the histogram for the moment, how are we integrating reflections on frame 44 when we have not loaded the full background for 44? How are we processing 22-44 but only integrating 30-44? 🤔

**Comment by graeme-winter** [comment]
Ah, this reports the spots who's shoeboxes are ended, not ones with _centroid_ on that image - 

```
 Frames: 2387 -> 2400

 Number of reflections
  Partial:     119
  Full:        107
  In ice ring: 0
  Integrate:   226
  Total:       226

 The following histogram shows the number of reflections predicted
 to have all or part of their intensity on each frame.

 2388 [5  ]: **
 2389 [13 ]: ******
 2390 [22 ]: **********
 2391 [45 ]: **********************
 2392 [56 ]: ***************************
 2393 [69 ]: **********************************
 2394 [81 ]: ****************************************
 2395 [80 ]: ***************************************
 2396 [83 ]: *****************************************
 2397 [81 ]: ****************************************
 2398 [79 ]: ***************************************
 2399 [81 ]: ****************************************
 2400 [135]: *******************************************************************

 Integrated     3 (sum) +     3 (prf) /    3 reflections on image 2393
 Integrated    15 (sum) +    15 (prf) /   16 reflections on image 2394
 Integrated    14 (sum) +    14 (prf) /   18 reflections on image 2395
 Integrated    12 (sum) +    12 (prf) /   15 reflections on image 2396
 Integrated    10 (sum) +    10 (prf) /   13 reflections on image 2397
 Integrated    12 (sum) +    12 (prf) /   14 reflections on image 2398
 Integrated    11 (sum) +    11 (prf) /   12 reflections on image 2399
 Integrated   107 (sum) +    35 (prf) /  135 reflections on image 2400
```

explains this then

**Comment by graeme-winter** [comment]
Resolved by 257ad048d72ae8c42c2fa8e4ece55d7ce63f69b4
